src/run/editor/mod.rs

```diff
@@ -662,26 +656,27 @@ impl Editor {
             return Ok(());
         }
 
-        if validate_comparison_name(&self.run, new) {
-            let position = self.run
-                .custom_comparisons()
-                .iter()
-                .position(|c| c == old)
-                .ok_or(())?;
+        match self.run.validate_comparison_name(new) {
```

Comment:
You don't do anything in the Err case, so you can just use ? instead of matching on it.

Reply:
Which also means that the return type of this function should have the proper Error type too.

Reply:
Also you have Result<T, ComparisonError> a lot. It may make sense to just provide a type def as ` ComparisonResult<T>`.